mfr/extensions/tabular/libs/h5py_scipy_tools.py

```diff
@@ -0,0 +1,60 @@
+import h5py
+import scipy.io
+
+from ..utilities import data_population, header_population
+from collections import OrderedDict
+
+
+def mat_v73(fp):
+    """Read and convert a mat v7.3+ file to JSON format using h5py
+    :param fp: File pointer object
+    :return: tuple of table headers and data
+    """
+
+    workbook = h5py.File(fp.name, 'r')
```

Comment:
As mentioned below, please check if this throws a certain exception and if `workbook` is `None`. Here is an example structure I have in mind just to provide some insights.

```python
try:
     workbook = h5py.File(fp.name, 'r')
except ...:
    # invalid version, re-raise it for trying another parser
     ...
except ...:
    # other parsing failure
    # log the issue and set workbook to None
    ...

if workbook:   
    # your code
    sheets = OrderedDict()
    variables = workbook.items()
    ...
    return sheets

# return an error message back to the user (not sure if it is here or handled by something else)
```